src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_chromaggus.cpp

```diff
@@ -294,7 +297,44 @@ class boss_chromaggus : public CreatureScript
     }
 };
 
+class go_chromaggus_lever : public GameObjectScript
+{
+    public:
+        go_chromaggus_lever() : GameObjectScript("go_chromaggus_lever") { }
+
+        struct go_chromaggus_leverAI : public GameObjectAI
+        {
+            go_chromaggus_leverAI(GameObject* go) : GameObjectAI(go), instance(go->GetInstanceScript()) { }
+
+            InstanceScript* instance;
+
+            bool GossipHello(Player* player) override
+            {
+                if (instance->GetBossState(DATA_CHROMAGGUS) != DONE && instance->GetBossState(DATA_CHROMAGGUS) != IN_PROGRESS)
+                {
+                    instance->SetBossState(DATA_CHROMAGGUS, IN_PROGRESS);
+                    if (Creature* creature = instance->GetCreature(DATA_CHROMAGGUS))
+                        creature->AI()->JustEngagedWith(player);
+                    
+                    if (GameObject* gate = me->FindNearestGameObject(GO_CHROMAGGUS_DOOR, 65.0f))
```

Comment:
we cant use the instance to find this gob? instead find near?

Reply:
*puts on obama voice*  yes we can!

Reply:
hi sucker <3

Reply:
like this? 30f6e0006a42f769be33824542be4cd99c577c83

Reply:
No. You add the door to GameObjectData in the instance script and acces it via  instance->GetGameObject(type);